File: pulumi/rds.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const auto_wp_db_cluster_pg = new aws.rds.ClusterParameterGroup(
  "auto-wp-db-cluster-pg",
  {
    description: "rds aurora mysql cluster parameter group for auto-wp",
    family: "aurora-mysql5.7",
    name: "auto-wp-db-cluster-pg",
    parameters: [
      {
        applyMethod: "immediate",
        name: "general_log",
        value: "1",
      },
      {
        applyMethod: "immediate",
        name: "slow_query_log",
        value: "1",
      },
    ],
  },
  {
    protect: true,
  }
);

const auto_wp_db_pg = new aws.rds.ParameterGroup(
  "auto-wp-db-pg",
  {
    description: "rds aurora mysql parameter group for auto-wp",
    family: "aurora-mysql5.7",
    name: "auto-wp-db-pg",
    parameters: [
      {
        applyMethod: "immediate",
        name: "slow_query_log",
        value: "1",
      },
      {
        applyMethod: "immediate",
        name: "general_log",
        value: "1",
      },
    ],
    tags: {
      Project: "auto-wp",
    },
  },
  {
    protect: true,
  }
);

const test_wp_db = new aws.rds.Cluster(
  "test-wp-db",
  {
    backupRetentionPeriod: 1,
    copyTagsToSnapshot: true,
    enableGlobalWriteForwarding: false,
    enableHttpEndpoint: false,
    engine: "aurora-mysql",
    engineMode: "provisioned",
    skipFinalSnapshot: true,
    // allowMajorVersionUpgrade: true,
    dbClusterParameterGroupName: auto_wp_db_cluster_pg.name,
    dbInstanceParameterGroupName: auto_wp_db_pg.name,
    enabledCloudwatchLogsExports: ["general", "error", "audit", "slowquery"],
    tags: {
      Project: "auto-wp",
    },
  },
  {
    protect: true,
  }
);

const test_wp_db_instance_1 = new aws.rds.ClusterInstance(
  "test-wp-db-instance-1",
  {
    autoMinorVersionUpgrade: true,
    clusterIdentifier: test_wp_db.id,
    copyTagsToSnapshot: false,
    engine: "aurora-mysql",
    instanceClass: "db.t2.small",
    monitoringInterval: 60,
    promotionTier: 1,
    dbParameterGroupName: auto_wp_db_pg.name,
    publiclyAccessible: false,
  },
  {
    protect: true,
  }
);

File: pulumi-py/rds.py
import pulumi
import pulumi_aws as aws

test_wp_db = aws.rds.Cluster("test-wp-db",
    backup_retention_period=1,
    copy_tags_to_snapshot=True,
    enable_global_write_forwarding=False,
    enable_http_endpoint=False,
    engine="aurora-mysql",
    engine_mode="provisioned",
    skip_final_snapshot=True,
    tags={
        "Project": "auto-wp",
    },
    opts=pulumi.ResourceOptions(protect=True))
test_wp_db_instance_1 = aws.rds.ClusterInstance("test-wp-db-instance-1",
    auto_minor_version_upgrade=True,
    cluster_identifier="test-wp-db",
    copy_tags_to_snapshot=False,
    engine="aurora-mysql",
    instance_class="db.t2.small",
    monitoring_interval=60,
    promotion_tier=1,
    publicly_accessible=False,
    opts=pulumi.ResourceOptions(protect=True))
